compiler/rustc_hir_analysis/src/check/check.rs

```diff
@@ -148,6 +149,53 @@ fn check_union_fields(tcx: TyCtxt<'_>, span: Span, item_def_id: LocalDefId) -> b
     true
 }
 
+/// Check that the unsafe fields do not need dropping.
+fn check_unsafe_fields(tcx: TyCtxt<'_>, span: Span, item_def_id: LocalDefId) -> bool {
+    let item_type = tcx.type_of(item_def_id).instantiate_identity();
+    if let ty::Adt(def, args) = item_type.kind() {
```

Comment:
You could de-nest this by writing:
```
let ty::Adt(def, args) = item_type.kind() else {
    span_bug!(span, "structs/enums must be ty::Adt, but got {:?}", item_type.kind());
    return;
}
/* rest of the function */
```

Reply:
`span_bug` diverges already, so no need to return.